librz/analysis/arch/ppc/ppc_il_ops.c

```diff
@@ -408,19 +409,68 @@ static RzILOpEffect *add_sub_op(RZ_BORROW csh handle, RZ_BORROW cs_insn *insn, b
 	return SEQ5(set_ops, set, set_carry, overflow, update_cr0);
 }
 
+static bool is_logical_compare(ut32 id) {
+	switch (id) {
+	case PPC_INS_CMPW:
+	case PPC_INS_CMPD:
+	case PPC_INS_CMPWI:
+	case PPC_INS_CMPDI:
+		return true;
+	default:
+		return false;
+	}
+}
+
 static RzILOpEffect *compare_op(RZ_BORROW csh handle, RZ_BORROW cs_insn *insn, const cs_mode mode) {
 	rz_return_val_if_fail(handle && insn, EMPTY());
 	ut32 id = insn->id;
-	const char *crX;
-	const char *rA;
-	const char *rB;
-	st64 imm;
+	const char *crX = NULL;
+	const char *rA = NULL;
+	const char *rB = NULL;
+	st64 imm = 0;
 
-	RzILOpPure *left;
-	RzILOpPure *right;
+	RzILOpPure *left = NULL;
+	RzILOpPure *right = NULL;
 
 	bool signed_cmp = false;
 
+#if CS_API_MAJOR > 4
+	// weird bug on cmp/cmpl in capstone v5
```

Comment:
Please point out what bug and link an issue if possible. Just in case someone needs to know.

Reply:
Also for the bug comments below as well please.

Reply:
@wargio and this one too